Core/Object Arts/Dolphin/MVP/MultiSelectListViewTest.cls

```diff
@@ -17,6 +17,116 @@ initializePresenter
 sortSelections
 	^#(49 50)!
 
+testNewSelectionsClickOutsideListWithModifiers
+	| event |
+	self setUpForSelectionTesting.
+	presenter selectionsByIndex: #(1).
+	(OrderedCollection new)
+		add: #(#control);
+		add: #(#shift);
+		add: #(#control #shift);
+		do: 
+				[:modifiers |
+				event := self mouseDownEventOnItem: 0 buttons: (modifiers copyWith: #left).
+				self assert: (presenter newSelectionsFromEvent: event) = #(1)].!
+
+testNewSelectionsCtrlClickOnSelectedItem
+	| event |
+	self setUpForSelectionTesting.
+	presenter selectionsByIndex: #(1 2).
+	event := self mouseDownEventOnItem: 1 buttons: #(#left #control).
+	self assert: (presenter newSelectionsFromEvent: event) = #(2).
+	presenter selectionsByIndex: #(1).
+	self assert: (presenter newSelectionsFromEvent: event) = #().!
+
+testNewSelectionsCtrlClickOnUnselectedItem
+	| event |
+	self setUpForSelectionTesting.
+	event := self mouseDownEventOnItem: 1 buttons: #(#left #control).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1).
+	presenter selectionsByIndex: #(2).
```

Comment:
Test of Ctrl+click to add a new selection item would be slightly better if the clicked item was 3+ so that there is a gap in the new selection range. This test as is would might pass even if the implementation mixed up ctrl and shift handling.